### ISSUE TITLE
Replace with checkbox component from react-science

### ISSUE THREAD

**Issue by lpatiny** [cheminfo/nmrium#2406]
Currently you can not click on the label of the checkbox but you need to click on the checkbox itself.

This can be solve using the react-science checkbox component

![2023-06-09 09 57 32](https://github.com/cheminfo/nmrium/assets/1484241/3a2c3a81-1531-4cae-8f81-f5015cd56d8a)

**Comment by hamed-musallam** [comment]
i think no need to use the checkbox from react-science, we can wrap the `IsotopesViewer` with a label or we can modify the IsotopesViewer to render it as a label.

https://github.com/cheminfo/nmrium/blob/main/src/component/elements/IsotopesViewer.tsx

**Comment by targos** [comment]
We don't "need" to use components from react-science, but the goal with that project is to build a good component library that is separate from nmrium to be more easily testable, discoverable, and to have more external contributions in order to improve it.

**Comment by wadjih-bencheikh18** [comment]
> Currently you can not click on the label of the checkbox but you need to click on the checkbox itself.

It's currently working in [https://dev.nmrium.org](https://dev.nmrium.org) , i checked the `FormikCheckBox` component and it works perfectly
@targos we can maybe update `Checkbox` component in `react-science` and make it more customizable (styling, positions ...)

**Comment by targos** [comment]
It was probably fixed after opening this issue, but the goal here is to replace uses of the [CheckBox](https://github.com/cheminfo/nmrium/blob/main/src/component/elements/CheckBox.tsx) component with the one from `react-science` (so in the end we can delete the component).
If you think we need to add features to the `react-science` component, please comment here with examples.

**Comment by wadjih-bencheikh18** [comment]
the feature that need to be add to use it in nmrium is checkbox style (label and input)
the style is used here in [FormikCheckBox](https://github.com/cheminfo/nmrium/blob/main/src/component/elements/formik/FormikCheckBox.tsx)

**Comment by targos** [comment]
Can you start by migrating all the other uses of the component (skip FormikCheckBox)?
Before adding props to the API, we should understand why they are needed. The decision can be to remove the option because it's not necessary.

**Comment by wadjih-bencheikh18** [comment]
We still have to
use react-science `CheckBox` in `FormikCheckBox`

**Comment by targos** [comment]
Yes, but no. I would like to push so Formik itself is replaced with something else (probably react-hook-form as that's what we have experience with). The Formik project is dead.

**Comment by hamed-musallam** [comment]
> The Formik project is dead

Are you sure? they still fix bugs and add features, the last release was on 17, Sep,2023.

we might consider moving to react-hook-form for potential performance improvements. However, at the moment, I don't see any significant issues with Formik that would warrant switching to a different package.

**Comment by targos** [comment]
They really only do a few bug fixes. There is no active maintenance.
Anyway, we dropped Formik in our projects a long time ago for other reasons. Performance is one of them (especially with large forms). DX is another reason.
I'm not suggesting to switch right now, though. Just that it's not necessary to migrate the existing components.

**Comment by hamed-musallam** [comment]
Yes, I agree we could switch right now and then subsequently migrate the other components.
